### PR TITLE
Mothur v1.42.3

### PR DESCRIPTION
**Pull request by wclose**
@bioconda/core this recipe updates the mothur version and should fix issues with gzipped files as part of the build. Please add me to the bioconda team. Thanks!

* [x] I have read the [guidelines for bioconda recipes](https://bioconda.github.io/contributor/guidelines.html).
* [ ] This PR adds a new recipe.
* [x] AFAIK, this recipe **is directly relevant to the biological sciences** (otherwise, please submit to the more general purpose [conda-forge channel](https://conda-forge.org/docs/)).
* [x] This PR updates an existing recipe.
* [ ] This PR does something else (explain below).

### CODE REVIEW

File: recipes/mothur/meta.yaml [hunk 1]
```diff
@@ -1,35 +1,48 @@
...
 package:
   name: mothur
   version: {{ version }}
 
 source:
-  url: https://github.com/mothur/mothur/archive/v{{ version }}.tar.gz
-  sha256: 6b61591dda289ac2d8361f9c1547ffbeeba3b9fbdff877dd286bad850bbd5539
+  - url: https://github.com/mothur/mothur/archive/v.{{ version }}.tar.gz
+    sha256: {{ sha256 }}
+  # - url: https://support.hdfgroup.org/ftp/HDF5/releases/hdf5-1.10/hdf5-1.10.5/src/hdf5-1.10.5.tar.gz
```
Comment (bgruening) [inline recipes/mothur/meta.yaml]: can you clean up this recipe a little bit?

File: recipes/mothur/1.42.1/build.sh [hunk 2]
```diff
@@ -0,0 +1,8 @@
+# the flags the new gxx activate script sets breaks the build, so unset them here
```
Comment (bgruening) [inline recipes/mothur/1.42.1/build.sh]: Do you think we should remove those old versions?

File: recipes/mothur/meta.yaml (hunk 1 above)
Comment (wclose) [reply recipes/mothur/meta.yaml]: Already planned on it. I was having issues with linking/libraries so those were remnants from my backup plan.

File: recipes/mothur/1.42.1/build.sh (hunk 2 above)
Comment (wclose) [reply recipes/mothur/1.42.1/build.sh]: Are you sure removal would be the best course? Not sure if that would mess with people that already have them implemented. I can apply the same fixes as this version and they 'should' compile as desired.

Comment (bgruening) [reply recipes/mothur/1.42.1/build.sh]: Removing the recipe does not mean removing the package. Lets formulate it the other way around, keeping the subdir means people want to maintain this particular version. So it gets rebuild against new python versoin etc. Is that really needed for this particual version?

Comment (wclose) [reply recipes/mothur/1.42.1/build.sh]: Ah, I understand what you're saying now. In that case, yeah, let's remove them. Should I just delete all the old subdirs then?

Comment (bgruening) [reply recipes/mothur/1.42.1/build.sh]: Yes :)

Comment (wclose) [reply recipes/mothur/1.42.1/build.sh]: Ok, it should be all set to go now. Thanks for your help!

**Comment by bgruening** [comment]
Thanks @wclose!

**Pull request merged**